middle_end/closure/closure.ml

```diff
@@ -772,7 +767,7 @@ let warning_if_forced_inline ~loc ~attribute warning =
 
 (* Generate a direct application *)
 
-let direct_apply ~backend fundesc ufunct uargs ~loc ~attribute =
+let direct_apply ~backend fundesc ufunct uargs ~loc ~attribute ~mutable_vars =
```

Comment:
Have you considered changing `direct_apply` to take the `env` parameter as a whole, instead of `~backend ~mutable_vars` separately? (I guess we need to move the `env` definition above here as a result, which is fine.)

Reply:
Makes sense, I amended as suggested.